numpy/_core/src/npysort/highway_argsort.dispatch.cpp

```diff
@@ -0,0 +1,63 @@
+#include "highway_qsort.hpp"
+#include "aquicksort.hpp"
+#include "common.hpp"
+#if VQSORT_ENABLED
+
+
+#define DISPATCH_VQSORT_u32(TYPE) \
+template<> void NPY_CPU_DISPATCH_CURFX(ArgQSort)(TYPE *arr, npy_intp *arg, npy_intp size) \
+{ \
+    hwy::K32V32* kv_pairs = new hwy::K32V32[size]; \
```

Comment:
These new arg quicksort methods require additional `O(N)` space. It needs  `size * sizeof(K32V32)  = size * 8` bytes. Can some please confirm if this is okay? AFAIK, quicksort methods are expected to be constant space. ping @charris?